applications/crossbar/src/api_util.erl

```diff
@@ -868,17 +868,24 @@ ensure_content_type(CT) -> CT.
 %% (the final module in the chain) accepts this verb parameter pair.
 %% @end
 %%--------------------------------------------------------------------
--spec does_resource_exist(cb_context:context()) -> boolean().
--spec does_resource_exist(cb_context:context(), list()) -> boolean().
+-spec does_resource_exist(cb_context:context()) -> {boolean(), cb_context:context()}.
+-spec does_resource_exist(cb_context:context(), list()) -> {boolean(), cb_context:context()}.
 does_resource_exist(Context) ->
     does_resource_exist(Context, cb_context:req_nouns(Context)).
 
 does_resource_exist(Context, [{Mod, Params}|_]) ->
     Event = api_util:create_event_name(Context, <<"resource_exists.", Mod/binary>>),
-    Responses = crossbar_bindings:map(Event, Params),
-    crossbar_bindings:any(Responses) and 'true';
-does_resource_exist(_Context, _ReqNouns) ->
-    'false'.
+    Responses = crossbar_bindings:map(Event, [Context, Params]),
```

Comment:
This should be `crossbar_bindings:map(Event, [Context | Params])` to match how we call `map/2` elsewhere. The crossbar endpoint modules must export different arity functions to handle different number of parameters from the URI.

*  `resource_exists() -> 'true'.` should become `resource_exists(Context) -> {'true', Context}.`
*  `resource_exists(_PathToken) -> 'true'.` should become `resource_exists(Context, _PathToken) -> {'true', Context}.`
*  `resource_exists(_PathToken1, _PathToken2) -> 'true'.` should become `resource_exists(Context, _PathToken1, _PathToken2) -> {'true', Context}.`

et cetera...